sklearn/utils/tests/test_pprint.py

```diff
@@ -303,16 +306,198 @@ def test_n_max_elements_to_show():
     assert  pp.pformat(gs) == expected
 
 
-def test_length_constraint():
-    # When repr is still too long, use bruteforce ellipsis
+def test_bruteforce_ellipsis():
+    # Check that the bruteforce ellipsis (used when the number of non-blank
+    # characters exceeds N_CHAR_MAX) renders correctly.
+
+    # Basic sanity check on very long strings.
     # repr is a very long line so we don't check for equality here, just that
-    # ellipsis has been done. It's not the ellipsis from before because the
-    # number of elements in the dict is only 1.
+    # ellipsis has been done. Note that this is a different ellipsis from
+    # N_MAX_ELEMENTS_TO_SHOW.
     vocabulary = {0: 'hello' * 1000}
     vectorizer = CountVectorizer(vocabulary=vocabulary)
     repr_ = vectorizer.__repr__()
     assert '...' in repr_
 
+    # make sure that ellipsis is done without considering the blank
+    # characters (see issue 13372). Also, this test covers the case when the
+    # left and right side of the ellipsis aren't on the same line.
+    numeric_features = ['age', 'fare']
+    numeric_transformer = Pipeline(steps=[
+        ('imputer', SimpleImputer(strategy='median')),
+        ('scaler', StandardScaler())])
+
+    categorical_features = ['embarked', 'sex', 'pclass']
+    categorical_transformer = Pipeline(steps=[
+        ('imputer', SimpleImputer(strategy='constant', fill_value='missing')),
+        ('onehot', OneHotEncoder(handle_unknown='ignore'))])
+
+    preprocessor = ColumnTransformer(
+        transformers=[
+            ('num', numeric_transformer, numeric_features),
+            ('cat', categorical_transformer, categorical_features)])
+
+    clf = Pipeline(steps=[('preprocessor', preprocessor),
+                        ('classifier', LogisticRegression(solver='lbfgs'))])
+
+    expected = """
```

Comment:
I had, btw, expected that with a configurable `N_CHAR_MAX`  you'd make this test case much simpler...

Reply:
you mean testing with an estimator that has a shorted repr?

I agree it'd be better (I'll do it if that's what you meant)

Reply:
Yes. Looking at the long, pathological cases give us a better sense of the user experience, but makes the tests harder to work with.

Reply:
updated, thanks it looks much better